File: src/lib/enums.ts
enum BackendEvent {
  ShowWindowSwitcher = "show-window-switcher",
  HideWindowSwitcher = "hide-window-switcher",
  ShowWindowSelector = "show-window-selector",
  HideWindowSelector = "hide-window-selector",
  ShowMonitorSelector = "show-monitor-selector",
  HideMonitorSelector = "hide-monitor-selector",
  ShowCalculator = "show-calculator",
  HideCalculator = "hide-calculator",
  SendProcesses = "send-processes"
}

enum Command {
  KillProcess = "kill_process",
  ShowWindowSelector = "show_window_selector",
  HideWindowSelector = "hide_window_selector",
  ShowMonitorSelector = "show_monitor_selector",
  HideMonitorSelector = "hide_monitor_selector",
  MinimizeWindow = "minimize_window",
  MaximizeWindow = "maximize_window",
  FocusWindow = "focus_window",
  CloseWindow = "close_window"
}

enum Operator {
  Add = "+",
  Subtract = "-",
  Multiply = "*",
  Divide = "/",
  Power = "^",
  Factorial = "!"
}

enum Function {
  Sine = "sin",
  Cosine = "cos"
}

enum Token {
  Point = ".",
  LeftParen = "(",
  RightParen = ")"
}

export { BackendEvent, Command, Operator, Function, Token };

File: src-tauri/src/libs/enums.rs
pub enum BackendEvent {
    ShowWindowSwitcher,
    HideWindowSwitcher,
    ShowWindowSelector,
    HideWindowSelector,
    ShowMonitorSelector,
    HideMonitorSelector,
    ShowCalculator,
    HideCalculator,
    SendProcesses,
}

impl BackendEvent {
    pub fn as_str(&self) -> &'static str {
        match self {
            BackendEvent::ShowWindowSwitcher => "show-window-switcher",
            BackendEvent::HideWindowSwitcher => "hide-window-switcher",
            BackendEvent::ShowWindowSelector => "show-window-selector",
            BackendEvent::HideWindowSelector => "hide-window-selector",
            BackendEvent::ShowMonitorSelector => "show-monitor-selector",
            BackendEvent::HideMonitorSelector => "hide-monitor-selector",
            BackendEvent::ShowCalculator => "show-calculator",
            BackendEvent::HideCalculator => "hide-calculator",
            BackendEvent::SendProcesses => "send-processes",
        }
    }
}

pub enum WindowLabel {
    WindowSwitcher,
    WindowSelector,
    MonitorSelector,
    Calculator,
}

impl WindowLabel {
    pub fn as_str(&self) -> &'static str {
        match self {
            WindowLabel::WindowSwitcher => "window-switcher",
            WindowLabel::WindowSelector => "window-selector",
            WindowLabel::MonitorSelector => "monitor-selector",
            WindowLabel::Calculator => "calculator",
        }
    }

    pub fn iter() -> impl Iterator<Item = WindowLabel> {
        vec![
            WindowLabel::WindowSwitcher,
            WindowLabel::WindowSelector,
            WindowLabel::MonitorSelector,
            WindowLabel::Calculator,
        ]
        .into_iter()
    }
}

#[derive(Clone)]
pub enum HotKeyName {
    WindowSwitcher,
    Calculator,
}

impl Copy for HotKeyName {}
